packages/polling-controller/src/PollingController.ts

```diff
@@ -49,27 +62,34 @@ function PollingControllerMixin<TBase extends Constructor>(Base: TBase) {
      * Starts polling for a networkClientId
      *
      * @param networkClientId - The networkClientId to start polling for
+     * @param options - The options used to group the polling events
      * @returns void
      */
-    startPollingByNetworkClientId(networkClientId: NetworkClientId) {
+    startPollingByNetworkClientId(
+      networkClientId: NetworkClientId,
+      options: Json = {},
+    ) {
       const innerPollToken = random();
-      if (this.#networkClientIdTokensMap.has(networkClientId)) {
-        const set = this.#networkClientIdTokensMap.get(networkClientId);
-        set?.add(innerPollToken);
+
+      const key = getKey(networkClientId, options);
+
+      const pollingGroupId = this.#pollingGroupIds.get(key);
```

Comment:
Is this right? I thought each value in `#pollingGroupIds` is set of poll tokens. It seems that `key` here is really the polling group ID.

Reply:
yeah currently the type you have returned for `getKey` is `PollingGroupId`, so presumably you'd use the `pollingGroupId` to get a `pollingGroup` or `pollingTokenGroup` or something like that

Reply:
yea really it should be `pollingTokenSet`, and `#pollingGroupIds` should be `#pollingTokenSets` or `#pollingTokensByPollingGroupId`

Reply:
> pollingTokenSet 
👍 

> #pollingTokenSets
I'd vote either this or just `pollingTokens` per @mcmire 's [comment](https://github.com/MetaMask/core/pull/1776#discussion_r1355582436)

Reply:
probably `pollingTokenSets` though since it would be kindof weird for `pollingTokenSet` to be a subset of `pollingTokens`?